simple_validate_magetab.pl

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env perl
+#
+# This script parsed MAGE-TAB for a given experiment idf
+#
+# It requires the conda perl-atlas-modules package to be installed.
+# Description:
+#     The script takes one parameter project accession if the idf wants to be generated for only one accession.
+
+use strict;
+use warnings;
+use File::Path;
+use 5.10.0;
+
+use Bio::MAGETAB::Util::Reader;
+use Bio::MAGETAB::Util::Writer::GraphViz;
+require Bio::MAGETAB::Util::Writer;
+use File::Find::Rule;
+use Try::Tiny;
+
+use Bio::MAGETAB::Util::Writer::IDF;
+use Bio::MAGETAB::Util::Writer::ADF;
+use Bio::MAGETAB::Util::Writer::SDRF;
+
+use File::Basename;
+
+my $project = '';
+if (@ARGV){
+  $project = $ARGV[0];
+}
+
+
+my $includeFiles = File::Find::Rule->file()->name('*.idf.tsv'); # search by file extensions
+if ($project ne ''){
+   $includeFiles = File::Find::Rule->file()
+     ->name($project . '*idf.tsv'); # search by file extensions
+
+}
+my $count_errors = 0;
+my $count_projects = 0;
+my @files = File::Find::Rule->or($includeFiles)->in('annotated-projects/');
+
+
+
+foreach (@files){
+
+  print('Validating project ' . $_ . "\n");
+  $count_projects = $count_projects + 1;
+
```

Comment:
I don't think that it is a good idea to embed all this file searching logic here. I would suggest that this goes as part of the shell logic and this script is kept cleaner.